skore/src/skore/_sklearn/_estimator/feature_importance_accessor.py

```diff
@@ -532,9 +593,58 @@ def _feature_permutation(
             # earlier.
             score = self._parent._cache[cache_key]
         else:
+            if not isinstance(self._parent.estimator_, Pipeline) or at_step == 0:
+                estimator = self._parent.estimator_
+                X_transformed = X_
+                if hasattr(estimator, "feature_names_in_"):
+                    feature_names = estimator.feature_names_in_
+                elif hasattr(X_transformed, "columns"):
+                    # Infer the column from the dataframe if it is one
+                    feature_names = X_transformed.columns.tolist()
+                else:
+                    feature_names = [
+                        f"Feature #{i}"
+                        for i in range(X_transformed.shape[1])  # type: ignore
+                    ]
```

Comment:
Since this code is almost the same as the one below starting at l.631, we could create a `get_feature_names(estimator, transformer=None)` where `transformer=None` when we have no pipeline or `at_step=0` and corresponds to `feature_eng` otherwise.